sdk/appcenter/src/main/java/com/microsoft/appcenter/utils/context/AuthTokenContext.java

```diff
@@ -131,47 +173,160 @@ private void updateAuthToken(String authToken, String homeAccountId) {
     }
 
     /**
-     * Check whether the user is new.
+     * Gets current authorization token.
      *
-     * @param newHomeAccountId account id of the logged in user.
-     * @return true if this user is not the same as previous, false otherwise.
+     * @return authorization token.
      */
-    private synchronized boolean isNewUser(String newHomeAccountId) {
-        return mHomeAccountId == null || !mHomeAccountId.equals(newHomeAccountId);
+    public synchronized String getAuthToken() {
+        List<AuthTokenHistoryEntry> history = getHistory();
+        if (history != null && history.size() > 0) {
+            return history.get(history.size() - 1).getAuthToken();
+        }
+        return null;
     }
 
     /**
-     * Clears info about the token.
+     * Gets current homeAccountId value.
+     *
+     * @return unique identifier of user.
      */
-    public synchronized void clearAuthToken() {
-        setAuthToken(null, null, null);
+    public synchronized String getHomeAccountId() {
+        List<AuthTokenHistoryEntry> history = getHistory();
+        if (history != null && history.size() > 0) {
+            return history.get(history.size() - 1).getHomeAccountId();
+        }
+        return null;
     }
 
     /**
-     * Cache auth token and account data to be used later on.
+     * Gets list of auth tokens validity info. It contains tokens and time when it was valid.
+     *
+     * @return auth token info.
+     * @see AuthTokenInfo
      */
-    public synchronized void cacheAuthToken() {
-        if (mStorage != null) {
-            updateAuthToken(mStorage.getToken(), mStorage.getHomeAccountId());
+    @NonNull
+    public synchronized List<AuthTokenInfo> getAuthTokenValidityList() {
```

Comment:
If a method get a list of auth tokens, it should rather be called getAllAuthTokens

Reply:
Renamed due to https://github.com/Microsoft/AppCenter-SDK-Android/pull/1014#discussion_r269941241
It returns a list of **not** just auth tokens. Will update javadoc a bit